### PR TITLE
Ensure random word doesn't contain unwanted chars

### PR DESCRIPTION
**Pull request by seanmetzgar**
- Adds regex to `random.word` to trim unwanted characters (Issue #1025)
- Adds `package-lock.json` to .gitignore file

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -17,6 +17,7 @@ reports/
 /npm-debug.log
 /coverage
 /.npm
+package-lock.json
```
Comment (hsw107) [inline .gitignore:20]: This should not be in the gitignore

File: lib/random.js [hunk 2]
```diff
@@ -203,7 +205,10 @@ function Random (faker, seed) {
     // randomly pick from the many faker methods that can generate words
     var randomWordMethod = faker.random.arrayElement(wordMethods);
     var result = faker.fake('{{' + randomWordMethod + '}}');
-    return faker.random.arrayElement(result.split(' '));
+    result = faker.random.arrayElement(result.split(' '));
+
+    // sanitize and retun word
```
Comment (hsw107) [inline lib/random.js:210]: Typo on "return"

**Review by hsw107 — COMMENTED** [review]
Thanks for doing this! Left some feedback that should hopefully help your PR go through!

File: .gitignore (hunk 1 above)
Comment (seanmetzgar) [reply .gitignore]: thanks, updated this with my latest commit 😄

File: lib/random.js (hunk 2 above)
Comment (seanmetzgar) [reply lib/random.js]: thanks, updated this with my latest commit 😄

File: .gitignore (hunk 1 above)
Comment (ohcibi) [reply .gitignore]: While in general this is true we haven't committed that file to master yet either. I think it's best to remove it from this PR. We will address the Lockfile issue in a separate branch.

/cc @Marak

Comment (hsw107) [reply .gitignore]: @ohcibi So the package-lock.json should be removed from the PR, but it still should **not** be added to the .gitignore, is that correct?

Comment (seanmetzgar) [reply .gitignore]: No problem, @ohcibi & @hsw107, I have removed the package-lock.json file in my latest commit, and left it out of .gitignore.

Comment (ohcibi) [reply .gitignore]: @hsw107 Let's wait for @Marak 's response. Im not sure. Maybe it is to not force anybody to either yarn or npm. I'll update the readme on this if necessary.

A general rule of thumb Im following myself is if in doubt I don't change anything unrelated to my pull request's topic.

File: lib/random.js [hunk 3]
```diff
@@ -166,6 +166,8 @@ function Random (faker, seed) {
    * @param {string} type
    */
   this.word = function randomWord (type) {
+    // regex statement used to remove unwanted characters from beginning/end of words
+    var wordSanitizerRegex = /^[\s\(\)\[\]\{\}\.,\-'"]+|[\s\(\)\[\]\{\}\.,\-'"]+$/g;
```
Comment (Moosh-be) [inline lib/random.js:170]: This list is perhaps "english centrics".
Is it difficult (or usefull or possible or untimely or premature ) to get  the list in locale config ?

Comment (hsw107) [reply lib/random.js:170]: Is there something about this list that is english centric?
I'm not sure what a locale config of making something more regex-friendly would look like.

**Pull request closed without merging**